types/db.go

```diff
@@ -29,8 +29,16 @@ type State struct {
 	VersionID  sql.NullInt64 `gorm:"index" json:"-"`
 	TFVersion  string        `gorm:"varchar(10)" json:"terraform_version"`
 	Serial     int64         `json:"serial"`
-	Lineage    string        `json:"lineage"`
-	Modules    []Module      `json:"modules"`
+	LineageID  uint          `gorm:"index" json:"-"`
```

Comment:
Is there a point in storing the LineageID instead of using a junction?

Reply:
For this point I don't really know which is a better solution so if you think a junction is better than a foreign key let's do that  :smile:

Reply:
My understanding is that gorm manages the foreign key automatically with the following line (`Lineage Lineage`) so it shouldn't be necessary to add the ID manually.

Reply:
After documentation investigating, you seems right!
I wasn't aware of that, I will change it! :wink:

Reply:
So I tested without LineageID field and it gave me that error: 
```[error] invalid field found for struct github.com/camptocamp/terraboard/types.State's field Lineage, need to define a valid foreign key for relations or it need to implement the Valuer/Scanner interface```
Moreover, without this field I am not sure I can still manually set foreign keys (essential for, for example, link the lineage with the state if it already exists during the insertion)

Maybe we should stay with a ```LineageID``` field in our case, you do you think about this?

Reply:
However, I will drop the Lineage field from the Plan and States structures which, after thinking about it, is useless and makes the model more complex. Initially, I had put it so that I could insert the lineage at the same time as them.

Reply:
I removed the Lineage field from the State structures which, after thinking about it, is useless and makes the model more complex. Initially, I had put it so that I could insert the lineage at the same time as them.

Reply:
I'm wondering if this could be solved by adding a `States []State` field to the `Lineage` struct (and also a `Plans []Plan` for the same reason), so that gorm understands that it is a one-to-many relationship. This would also allow to list all States and Plans associated with a Lineage directly from the Lineage struct.

Reply:
Nevermind, it's already there.